frontend/frontend-nav-menu.php

```diff
@@ -107,7 +115,7 @@ public function wp_get_nav_menu_items( $items ) {
 				/** This filter is documented in include/switcher.php */
 				$options = apply_filters( 'pll_the_languages_args', $options ); // Honor the filter here for 'show_flags', 'show_names' and 'dropdown'.
 
-				$switcher = new PLL_Switcher();
+				$switcher = PLL_Switcher::create( $this->polylang );
```

Comment:
Do we really need to call on the factory method here? Since we are in the `PLL_Frontend_Nav_Menu`, couldn't we always instantiate a `PLL_Frontend_Switcher` instead?

Reply:
Good question 🤔 Indeed we are supposed to be already in a frontend context.